mysql-test/suite/innodb/t/mem_pressure.test

```diff
@@ -0,0 +1,32 @@
+--source include/have_debug.inc
+--source include/linux.inc
+--source include/not_embedded.inc
+--source include/have_innodb.inc
+--source include/have_sequence.inc
+
+--echo #
+--echo # MDEV-24670 avoid OOM by linux kernel co-operative memory management
+--echo #
+
+set @save_dbug=@@debug_dbug;
+
+CREATE TABLE t1 (t TEXT) ENGINE=InnoDB;
+INSERT INTO t1 SELECT CONCAT(REPEAT('junk text', 500), CAST(seq AS CHAR)) FROM seq_1_to_1000;
```

Comment:
This will fill the buffer pool with undo log pages due to row-level undo logging. We should enable the bulk insert interface to fill the buffer pool mainly with the pages of the table `t1`. Therefore, it can happen that before and after the `buf_pool_t::garbage_collect()`, the buffer pool is filled with undo log pages.

Because we are using debug instrumentation anyway, we can also make use of `innodb_limit_optimistic_insert_debug` and write considerably less redo log.

We’d better wait that all undo logs have been purged before we start the test. Before I added that, the test would fail after a couple of dozen repetitions.
```diff
diff --git a/mysql-test/suite/innodb/t/mem_pressure.test b/mysql-test/suite/innodb/t/mem_pressure.test
index 2d140fb22ad..60f5acc291d 100644
--- a/mysql-test/suite/innodb/t/mem_pressure.test
+++ b/mysql-test/suite/innodb/t/mem_pressure.test
@@ -9,17 +9,27 @@
 --echo #
 
 set @save_dbug=@@debug_dbug;
-
-CREATE TABLE t1 (t TEXT) ENGINE=InnoDB;
-INSERT INTO t1 SELECT CONCAT(REPEAT('junk text', 500), CAST(seq AS CHAR)) FROM seq_1_to_1000;
-
+set @save_limit=@@GLOBAL.innodb_limit_optimistic_insert_debug;
+SET GLOBAL innodb_max_purge_lag_wait=0;
+
+CREATE TABLE t1 (a INT PRIMARY KEY) ENGINE=InnoDB;
+SET GLOBAL innodb_limit_optimistic_insert_debug=2;
+SET unique_checks=0, foreign_key_checks=0;
+INSERT INTO t1 SELECT * FROM seq_1_to_1000;
+SET GLOBAL innodb_limit_optimistic_insert_debug=@save_limit;
 DROP TABLE t1;
-SELECT CAST(VARIABLE_VALUE AS INTEGER) INTO @dirty_prev FROM INFORMATION_SCHEMA.GLOBAL_STATUS WHERE VARIABLE_NAME='Innodb_buffer_pool_pages_dirty';
+SET unique_checks=1, foreign_key_checks=1;
+
+SELECT CAST(VARIABLE_VALUE AS INTEGER) INTO @dirty_prev
+FROM INFORMATION_SCHEMA.GLOBAL_STATUS
+WHERE VARIABLE_NAME='Innodb_buffer_pool_pages_dirty';
 
 set debug_dbug="d,trigger_garbage_collection";
 SET GLOBAL innodb_buffer_pool_size=@@innodb_buffer_pool_size;
 
-SELECT VARIABLE_VALUE < @dirty_prev AS LESS_DIRTY_IS_GOOD FROM INFORMATION_SCHEMA.GLOBAL_STATUS WHERE VARIABLE_NAME='Innodb_buffer_pool_pages_dirty';
+SELECT CAST(VARIABLE_VALUE AS INTEGER) < @dirty_prev AS LESS_DIRTY_IS_GOOD
+FROM INFORMATION_SCHEMA.GLOBAL_STATUS
+WHERE VARIABLE_NAME='Innodb_buffer_pool_pages_dirty';
 
 let SEARCH_FILE= $MYSQLTEST_VARDIR/log/mysqld.1.err;
 let SEARCH_PATTERN= InnoDB: Memory pressure event freed;
@@ -27,6 +37,4 @@ let SEARCH_PATTERN= InnoDB: Memory pressure event freed;
 
 set debug_dbug=@save_dbug;
 
---echo #
 --echo # End of 10.11 tests
---echo #
```
The test as modified above passed my local test:
```
innodb.mem_pressure 'innodb'             w22 [ 1000 pass ]    139
--------------------------------------------------------------------------
The servers were restarted 0 times
Spent 5053.123 of 291 seconds executing testcases

Completed: All 42000 tests were successful.
```